webui/react/src/pages/Cluster.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useCallback } from 'react';
+
+import Grid from 'components/Grid';
+import Page from 'components/Page';
+import ResourceChart from 'components/ResourceChart';
+import Spinner from 'components/Spinner';
+import AgentsCtx from 'contexts/Agents';
+import emptyMessage from 'styles/emptyMessage.module.scss';
+import { Resource } from 'types';
+import { categorize } from 'utils/data';
+
+import css from './Cluster.module.scss';
+
+const Cluster: React.FC = () => {
+  const agents = AgentsCtx.useStateContext();
+
+  const getResources = useCallback(() => {
```

Comment:
suggestion: use `useMemo` instead so you can just set the resources directly instead of having to make another `getResources()` call.

```
const resourceDict = useMemo(() => {
  ...
  return ...;
}, [ agents ]);
```